src/com/wsscode/pathom/fulcro/network.cljs

```diff
@@ -187,24 +187,29 @@
                                (cond-> response
                                  id-param (assoc ::fp/tempids {(val id-param) (get response (graphql-response-key (key id-param)))}))))})}))
 
-(defn graphql-network [url]
-  (fn-network
-    (fn [this edn ok error]
-      (go
-        (try
-          (let [edn      (-> edn
-                             p/query->ast
-                             (p/elide-ast-nodes #{::pp/profile})
-                             p/ast->query)
-                query    (pg/query->graphql edn {::pg/js-name (comp pg/camel-case name)})
-                response (<? (fetch/request-async {::http/url         url
-                                                   ::http/method      ::http/post
-                                                   ::http/as          ::http/json
-                                                   ::http/form-params {:query query}}))
-                {:keys [data errors]} (::http/body response)]
-            (ok (graphql-response-parser {::p/entity data} edn)))
-          (catch :default e
-            (error e)))))))
+(defn graphql-network
+      ([url]
+        (graphql-network url {}))
+      ([url options]
+        (fn-network
+          (fn [this edn ok error]
+              (go
+                (try
+                  (let [edn (-> edn
+                                p/query->ast
+                                (p/elide-ast-nodes #{::pp/profile})
+                                p/ast->query)
+                        query (pg/query->graphql edn {::pg/js-name (comp pg/camel-case name)})
+                        response (<? (fetch/request-async (merge
```

Comment:
I have another idea on how to extend this, the merging is nice but that blocks using these options to change something else about the process in the future, what if instead the user can provide some fn that will update the request, this way the user could also read the defaults, this merge would be turned in:

```clojure
(cond-> {::http/url url ...} update-http-request update-http-request)
```

Then the user can do:

```clojure
(pfn/graphql-network url {::pfn/update-http-request #(assoc :headers {"myheader" "value"})}
```

What you think?

Reply:
I like it more. Refactoring.